src/osemgrep/cli_interactive/Interactive_subcommand.ml

```diff
@@ -796,24 +855,26 @@ let execute_command (state : state) =
    Care must be taken to ensure multiple threads don't mess with each other.
 *)
 let spawn_thread_if_turbo state =
-  if state.turbo then
-    Thread.create
-      (fun _ ->
-        let cur_line = get_current_line state in
-        let pat_opt = parse_pattern_opt cur_line state in
-        match (cur_line, pat_opt) with
-        | "", _ ->
-            (* When we go back to the empty line, reset the view to default. *)
-            should_refresh := true;
-            state.file_zipper :=
-              Framed_zipper.empty_with_max_len (height_of_files state.term)
-        | _, None -> ()
-        | _, Some pat ->
-            let new_iformula = IPat (pat, true) in
-            let file_zipper = matches_of_new_iformula new_iformula state in
-            state.file_zipper := file_zipper;
-            should_refresh := true)
-      ()
+  (* Let's only spawn a new turbo thread if we don't have more events in
+     our queue.
```

Comment:
But what if the next event is a noop? Then no thread will ever be created for it?
Maybe add a comment why this situation can't happen.

Reply:
oh, good point. actually, this would mess us up in that instance. i changed it so we only don't spawn if the next is guaranteed to create another turbo thread